server/src/routes/auth.ts

```diff
@@ -0,0 +1,111 @@
+import bcrypt from 'bcryptjs';
+import { Router } from 'express';
+import * as yup from 'yup';
+import { User } from '../entity/User';
+
+const authRouter = Router();
+
+yup.ValidationError;
+
+const registerSchema = yup.object().shape({
+  user_name: yup
+    .string()
+    .min(4)
+    .max(25),
+  user_email: yup
+    .string()
+    .email()
+    .max(255),
+  user_password: yup
+    .string()
+    .min(6)
+    .max(255),
+});
+
+const loginSchema = yup.object().shape({
+  user_email: yup
+    .string()
+    .email()
+    .max(255),
+  user_password: yup
+    .string()
+    .min(6)
+    .max(255),
+});
+
+authRouter.post('/register', async (req, res) => {
+  const { user_name, user_email, user_password } = req.body;
+  try {
+    registerSchema.validateSync(
+      {
+        user_name,
+        user_email,
+        user_password,
+      },
+      { abortEarly: false },
+    );
+
+    const userByEmail = await User.findOne({ user_email });
```

Comment:
this could be ran in parallel with `Promise.all` for some extra efficiency :rocket:

Reply:
i.e
```typescript
[userByEmail, userById] = await Promise.all([
  user.findOne(...
  user.findOne(...
]);
...
```